clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -19,6 +19,21 @@
  * If a valid partition number is specified that partition will be used when sending the record. If no partition is
  * specified but a key is present a partition will be chosen using a hash of the key. If neither key nor partition is
  * present a partition will be assigned in a round-robin fashion.
+ * <p>
+ * The record also has an associated timestamp. If user did not provide a timestamp, the producer will stamp the record
+ * with its current time. The timestamp eventually used by Kafka depends on the timestamp type configured for the topic.
+ * <li>
+ * If the topic is configured to use {@link org.apache.kafka.common.record.TimestampType#CreateTime CreateTime}
+ * the timestamp in the producer record will be used by broker.
+ * </li>
+ * <li>
+ * If the topic is configured to use {@link org.apache.kafka.common.record.TimestampType#LogAppendTime LogAppendTime}
+ * the timestamp in the producer record will be overwritten by broker with broker local time when broker append the
+ * message to its log.
+ * </li>
+ * <p>
+ * In either of the above case, the timestamp that has actually been used will be returned to user in
```

Comment:
`In either of the cases above, the timestamp that has actually been used will be returned to the user in`